doc/CrudSamples.java

```diff
@@ -0,0 +1,388 @@
+        // They should be of small length -- maximum a few MB -- because they are
+        // replicated to and from the server in a way which doesn't allow for resuming
+        // an upload or download.
+        //
+        // Attachments are stored via the `attachments` getters/setters on a DocumentRevision
+        // object. This is a map of attachments, keyed by attachment name.
+        //
+        // To add an attachment to a document, just add (or overwrite) the attachment
+        // in the `attachments` map:
+
+        // create a body
+        rev.setBody(DocumentBodyFactory.create(json));
+
+        // create an UnsavedFileAttachment: the constructor takes
+        // a File object on disk and a MIME type
+        UnsavedFileAttachment att1 = new UnsavedFileAttachment(
+                new File("/path/to/image.jpg"), "image/jpeg");
+
+        // As with the document body, you can replace the attachments
+        rev.setAttachments(new HashMap<String, Attachment>());
+
+        // Or just add or update a single one:
+        // (because the getter will always return the underlying map and not a copy)
+        rev.getAttachments().put(att1.name, att1);
+
+        DocumentRevision saved = ds.database().create(rev);
+
+        // When creating new attachments, use `UnsavedFileAttachment` for data you already have on
+        // disk. Use
+        // `UnsavedStreamAttachment` when you have data which comes from an InputStream or is
+        // already in
+        // memory.
+
+        att1 = new UnsavedFileAttachment(
+                new File("/path/to/image.jpg"), "image/jpeg");
+
+        byte[] imageData = new byte[256];
+        // ByteArrayInputStream adapts imageData to be used as a stream
+        UnsavedStreamAttachment att2 = new UnsavedStreamAttachment(
+                new ByteArrayInputStream(imageData), "cute_cat.jpg", "image/jpeg");
+
+        rev = new DocumentRevision();
+        rev.getAttachments().put(att1.name, att1);
+        rev.getAttachments().put(att2.name, att2);
+        saved = ds.database().create(rev);
+
+        // To read an attachment, get the `SavedAttachment` from the `attachments`
+        // map. Then use `getInputStream()` to read the data:
+
+        retrieved = ds.database().read("myDoc");
+        Attachment att = retrieved.getAttachments().get("cute_cat.jpg");
+        InputStream is = att.getInputStream();
+
+        // get all bytes in memory if required
+        byte[] data = new byte[(int) att.length];
+        is.read(data);
+
+        // To remove an attachment, remove it from the `attachments` map:
+
+        retrieved = ds.database().read("myDoc");
+        retrieved.getAttachments().remove("cute_cat.jpg");
+        updated = ds.database().update(retrieved);
+
+        // To remove all attachments, set the `attachments` property to an empty map
+        // or `null`:
+
+        updated.setAttachments(null);
```

Comment:
Given that the previous shows the `ds.database().update()` required to persist this change, this probably should as well.